pos_picking_load/static/src/js/pos_picking_load.js

```diff
@@ -0,0 +1,329 @@
+                }
+                else{
+                    self.$el.find('span.button.validate').hide();
+                }
+
+            }).fail(function (error, event){
+                if (parseInt(error.code) === 200) {
+                    // Business Logic Error, not a connection problem
+                    self.pos_widget.screen_selector.show_popup(
+                        'error-traceback', {
+                            message: error.data.message,
+                            comment: error.data.debug
+                        });
+                }
+                else{
+                    self.pos_widget.screen_selector.show_popup('error',{
+                        message: _t('Connection error'),
+                        comment: _t('Can not execute this action because the POS is currently offline'),
+                    });
+                }
+                event.preventDefault();
+            });
+        },
+
+        search_pickings: function(query) {
+            var self = this;
+            var pickingModel = new instance.web.Model(this.model);
+            return pickingModel.call('search_pickings_for_pos', [query || ''])
+            .then(function (result) {
+                self.render_list(result);
+            }).fail(function (error, event){
+                if (parseInt(error.code) === 200) {
+                    // Business Logic Error, not a connection problem
+                    self.pos_widget.screen_selector.show_popup(
+                        'error-traceback', {
+                            message: error.data.message,
+                            comment: error.data.debug
+                        }
+                    );
+                }
+                else{
+                    self.pos_widget.screen_selector.show_popup('error',{
+                        message: _t('Connection error'),
+                        comment: _t('Can not execute this action because the POS is currently offline'),
+                    });
+                }
+                event.preventDefault();
+            });
+        },
+
+        on_click_picking: function(event){
+            this.load_picking(parseInt(event.target.parentNode.dataset.pickingId));
+        },
+
+        render_list: function(pickings){
+            var self = this;
+            var contents = this.$el[0].querySelector('.picking-list-contents');
+            contents.innerHTML = "";
+            for (var i = 0, len = pickings.length; i < len; i++){
+                var picking = pickings[i];
+                var pickingline_html = QWeb.render('LoadPickingLine',
+                    {widget: this, picking:pickings[i]});
+                var pickingline = document.createElement('tbody');
+                pickingline.innerHTML = pickingline_html;
+                pickingline = pickingline.childNodes[1];
+                pickingline.addEventListener('click', this.on_click_picking);
+                contents.appendChild(pickingline);
```

Comment:
it's better (for the perfs) if you add all nodes in one DOM call.
you may also add the event listener on the collection instead of each row